### PR TITLE
[usage] Upload report with more details

### PR DESCRIPTION
**Pull request by easyCZ**
## Description

Reuses existing UsageReport struct to upload more details as part of the report.

## Related Issue(s)

Fixes #

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-preview

### CODE REVIEW

**Comment by roboquat** [comment]
@easyCZ: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: components/usage/pkg/apiv1/report-generator.go [hunk 1]
```diff
@@ -16,23 +17,6 @@ import (
 	"gorm.io/gorm"
 )
 
-type InvalidSession struct {
```
Comment (easyCZ) [inline components/usage/pkg/apiv1/report-generator.go:19]: This is moved to the `contentservice` package.

File: components/usage/pkg/apiv1/usage_test.go [hunk 2]
```diff
@@ -327,7 +327,7 @@ func TestReportGenerator_GenerateUsageReportTable(t *testing.T) {
 		return Must(db.NewVarcharTimeFromStr(timestampAsStr))
 	}
 	type Expectation struct {
-		custom       *func(t *testing.T, report *UsageReport)
+		custom       func(t *testing.T, report UsageReport)
```
Comment (easyCZ) [inline components/usage/pkg/apiv1/usage_test.go:330]: A function is already a pointer, no need to take the pointer again.

File: components/usage/pkg/contentservice/usage_report.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+	"github.com/gitpod-io/gitpod/usage/pkg/db"
+	"time"
+)
+
+type InvalidSession struct {
+	Reason  string
+	Session db.WorkspaceInstanceForUsage
+}
+
+type UsageReport struct {
```
Comment (easyCZ) [inline components/usage/pkg/contentservice/usage_report.go:17]: We would not upload this whole object to contentservice.

File: components/usage/pkg/contentservice/usage_report.go [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+	GenerationTime time.Time `json:"generation_time"`
```
Comment (andrew-farries) [inline components/usage/pkg/contentservice/usage_report.go:18]: It's a shame that we have a mix of camel and snake case for the field names here. Is that something that we can change now?

**Review by andrew-farries — APPROVED** [review]

File: components/usage/pkg/contentservice/usage_report.go (hunk 4 above)
Comment (easyCZ) [reply components/usage/pkg/contentservice/usage_report.go:18]: Yep, I think we still can

Comment (easyCZ) [reply components/usage/pkg/contentservice/usage_report.go:18]: https://github.com/gitpod-io/gitpod/pull/12511

**Pull request merged**